env/env.cc

```diff
@@ -95,15 +95,19 @@ void LogFlush(Logger *info_log) {
   }
 }
 
-void Log(Logger* info_log, const char* format, ...) {
+static void Logv(Logger *info_log, const char* format, va_list ap) {
   if (info_log && info_log->GetInfoLogLevel() <= InfoLogLevel::INFO_LEVEL) {
-    va_list ap;
-    va_start(ap, format);
     info_log->Logv(InfoLogLevel::INFO_LEVEL, format, ap);
-    va_end(ap);
   }
 }
 
+void Log(Logger* info_log, const char* format, ...) {
+  va_list ap;
+  va_start(ap, format);
+  Logv(info_log, format, ap);
+  va_end(ap);
+}
+
```

Comment:
Actually, is the log level check you want already on line 114-116 below? I think that should be applied to the `shared_ptr<Logger>` overloads. Am I missing something?

Reply:
Even if this is true, we can still take this diff - just remove the redundant checks in the `*v()` functions.

Reply:
That's the default implementation of Logv - if you override that rather than the old overload that doesn't take the log level you'd have to add the check in yourself.

Reply:
understood thanks. wish that function weren't virtual... :/